File: design-pattern/scala-java/src/main/java/io/loustler/massemble/designpattern/example/car/BuilderDemo.java
package io.loustler.massemble.designpattern.example.car;

public class BuilderDemo {
  public static void main(String[] args) {
    final JCar myCar = JCar.builder()
      .withType(JType.SEDAN)
      .withSeats(4)
      .withEngine(JEngine.V6)
      .withTransmission(JTransmission.AUTO)
      .withGpsNavigator(JGPSNavigator.T_MAP)
      .build();

    myCar.drive();
  }
}

File: design-pattern/scala-java/src/main/scala/io/loustler/massemble/designpattern/builder/BuilderDemo.scala
package io.loustler.massemble.designpattern.builder

object BuilderDemo {

  def main(args: Array[String]): Unit = {
    val car: Car = Car(engine = Engine.V6, navigator = GPSNavigator.TMap)

    assert(car.`type` == Type.Sedan)
    assert(car.engine == Engine.V6)
    assert(car.seats == 4)
    assert(car.transmission == Transmission.Auto)
    assert(car.navigator == GPSNavigator.TMap)

    car.drive()
  }
}
